### PR TITLE
Add support for ed25519 ssh keys

### PR DESCRIPTION
**Pull request by muhittink**
Since there are stronger ssh keys to protect a ssh connection, please add ed25519 as one of the defaults.

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -221,7 +221,7 @@ func searchSSHIdentity() (string, error) {
 	}
```
Comment (majst01) [inline cmd/root.go:221]: searchSSHIdentity is used to search for a suitable ssh identity file name, not the content. So i think we should reuse this function for the filename in machine.go on both places: machineCreate and machine Console

File: cmd/machine.go [hunk 2]
```diff
@@ -457,7 +459,17 @@ func machineCreateRequest() (*metalgo.MachineCreateRequest, error) {
 	}
 
 	if len(sshPublicKeyArgument) == 0 {
-		sshPublicKeyArgument, _ = readFromFile("~/.ssh/id_rsa.pub")
+		currentUser, err := user.Current()
+		homeDir := currentUser.HomeDir
+		defaultDir := filepath.Join(homeDir, "/.ssh/")
+		keys := []string{"id_ed25519.pub", "id_rsa.pub", "id_dsa.pub"}
```
Comment (majst01) [inline cmd/machine.go:465]: We should put either the allowed public key file names in a named constant, or take all files with *.pub suffix from the .ssh directory

File: cmd/root.go (hunk 1 above)
Comment (muhittink) [reply cmd/root.go:223]: Hi Stefan, you are right, I did some changes

File: cmd/root.go [hunk 3]
```diff
@@ -214,16 +216,15 @@ func initPrinter() {
...
 }
 
-func searchSSHIdentity() (string, error) {
+func searchSSHKey(sshKeys [3]string) (string, error) {
```
Comment (majst01) [inline cmd/root.go:219]: do not specify slice length here

File: cmd/root.go [hunk 4]
```diff
@@ -26,11 +26,13 @@ const (
...
-	printer   Printer
-	detailer  Detailer
-	driverURL string
-	driver    *metalgo.Driver
+	ctx                   Context
+	printer               Printer
+	detailer              Detailer
+	driverURL             string
+	driver                *metalgo.Driver
+	defaultSSHPublicKeys  = [...]string{"id_ed25519.pub", "id_rsa.pub", "id_dsa.pub"}
```
Comment (majst01) [inline cmd/root.go:34]: just specify the possibleKeyformats and add ".pub" for the public keys, so only one slice

File: cmd/machine.go [hunk 5]
```diff
@@ -457,9 +457,10 @@ func machineCreateRequest() (*metalgo.MachineCreateRequest, error) {
 	}
 
 	if len(sshPublicKeyArgument) == 0 {
-		sshPublicKey, _ := searchSSHKey(defaultSSHPublicKeys)
+		sshKey, _ := searchSSHKey()
```
Comment (majst01) [inline cmd/machine.go:460]: nooo, always check for errors

Comment (muhittink) [reply cmd/machine.go]: :D

File: cmd/machine.go [hunk 6]
```diff
@@ -457,7 +457,16 @@ func machineCreateRequest() (*metalgo.MachineCreateRequest, error) {
 	}
 
 	if len(sshPublicKeyArgument) == 0 {
-		sshPublicKeyArgument, _ = readFromFile("~/.ssh/id_rsa.pub")
+		var err error
+		sshKey, err := searchSSHKey()
+		if err != nil {
+			return nil, err
+		}
+		sshKey += ".pub"
```
[the remark this answers is not in the document]
Comment (muhittink) [reply cmd/machine.go]: that is more clear, thx

**Pull request merged**